job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/publisher/RundeckContext.groovy

```diff
@@ -36,4 +37,21 @@ class RundeckContext implements Context {
     void shouldFailTheBuild(boolean shouldFailTheBuild = true) {
         this.shouldFailTheBuild = shouldFailTheBuild
     }
+
+    //Requires Rundeck plugin 3.4+
```

Comment:
As the messages says, you need a jobManagement field in your class because the annotation needs a JobManagement instance to perform the check. You get that when the context inherits from AbstractContext.